en/2-wallet/README.md

```diff
@@ -14,3 +14,259 @@
+So how do we actually generate a wallet?  First we generate a private key, which is simply a 64 character hexadecimal string. This represents a number between the range 0 and 2^256 (1.15792089e77). From this number, the rest of your “Account” information is derived. For our purposes an account will consist of your Private Key, WIF (Wallet Import Format), public key, and address.
+
+The first challenge of any wallet software is deriving the account from the private key. We’ll go briefly go into more details of the 3 other keys. If you’re already familiar with address generation from another blockchain feel free to skip ahead, as Neo basically follows the same method as all others.
+
+### WIF
+The WIF is relatively to understand. In practice a private key can end up looking something like this…
+
+0C28FCA386C7A227600B2FE50B7CAEEC86D3BF1FBE471BE89827E19D72AA1D 
+
+It would be nice to have something that is at least a little bit more human readable, so we can convert the private key into the WIF otherwise known as the wallet import format
+
+5HueCGU8rMjxEXxiPuD5BDku4MkFqeZyd4dZ1jvhTVqvbTLvyTJ
+
+which although still not entirely readable is certainly better than the original string. WIF also has some basic error checking so that when you send to an address denominated by WIF format you are more likely to catch an error. The conversion from the raw private key to the wif format was done via a BASE 58 check encoding algorithm.
+
+## Base 58 check encoding
+Base 58 is similar to the common base 64 encoding scheme exept that it removes non alphanumeric characters as well as characters that might look similar to each other to the human eye. For example 0 (zero), O (capital o), I (capital i) and l (lower case L) are all omitted from the base 58 encoding scheme. The full list of available characters in NEO's base58 encoding is 123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz.
+
+A full implementation of NEO's check encoding written in go can be seen below
+```
+func b58checkencode(ver uint8, b []byte) (s string) {
+	/* Prepend version */
+	bcpy := append([]byte{ver}, b...)
+
+	/* Create a new SHA256 context */
+	sha256_h := sha256.New()
+
+	/* SHA256 Hash #1 */
+	sha256_h.Reset()
+	sha256_h.Write(bcpy)
+	hash1 := sha256_h.Sum(nil)
+
+	/* SHA256 Hash #2 */
+	sha256_h.Reset()
+	sha256_h.Write(hash1)
+	hash2 := sha256_h.Sum(nil)
+
+	/* Append first four bytes of hash */
+	bcpy = append(bcpy, hash2[0:4]...)
+
+	/* Encode base58 string */
+	s = b58encode(bcpy)
+
+	/* For number of leading 0's in bytes, prepend 1 */
+	for _, v := range bcpy {
+		if v != 0 {
+			break
+		}
+		s = "1" + s
+	}
+
+	return s
+}
+```
+The steps to perform the check encoding can be broken down as follows
+1. Prepend the version byte
+2. Double Hash the resulting hex using SHA256
+3. Append the first bytes of the hash to the prepended version
+4. Convert the hex with prepended version and appended checksum to base58.
+5. If any leading zeros in the bytes attach 1
+
```

Comment:
There is a error here.**checksum** is the first four bytes of the hash,not the first one.